src/mds/CDentry.cc

```diff
@@ -392,31 +392,6 @@ bool CDentry::is_freezing() const
   return dir->is_freezing();
 }
 
-void CDentry::decode_replica(bufferlist::iterator& p, bool is_new)
-{
-  __u32 nonce;
-  decode(nonce, p);
-  replica_nonce = nonce;
-  
-  decode(first, p);
-
-  inodeno_t rino;
-  unsigned char rdtype;
-  decode(rino, p);
-  decode(rdtype, p);
-  lock.decode_state(p, is_new);
-
-  bool need_recover;
-  decode(need_recover, p);
-
-  if (is_new) {
-    if (rino)
-      dir->link_remote_inode(this, rino, rdtype);
-    if (need_recover)
-      lock.mark_need_recover();
-  }
-}
-
```

Comment:
Instead of moving code to MDCache.cc. I prefer to move code from MDCache.cc to CDir/CInode/CDentry.cc

Reply:
I'm not sure if we can make all these encode/decode method to the member function of 
CDir/CInode/CDentry
I will try,  if we can do, it's more reasonable!

Reply:
zheng, i have looked into it, we can do it for encode method, but can't  for decode method, because decode method is creating CDir/CInode/CDentry itself..

Reply:
we can pass  struct_v to the decode function